### ISSUE TITLE
`no-unused-expressions` reported on `{' '}`

### ISSUE THREAD

**Issue by karlhorky** [mdx-js/eslint-mdx#548]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

eslint-plugin-mdx@3.1.5

### Steps to reproduce

1. Set up `eslint-plugin-mdx@3.1.5`
2. Lint the following file

`index.mdx`

```mdx
Before{' '}after
```

Observe that a problem is reported on the `{' '}`:

![Screenshot 2024-08-07 at 18 28 31](https://github.com/user-attachments/assets/8e4e9dc4-c4f9-4d8f-90c6-b30d121d043b)

```
Expected an assignment or function call and instead saw an expression. eslint no-unused-expressions
```

**Possibly related**

- https://github.com/mdx-js/eslint-mdx/issues/384

### Expected behavior

No problem is reported

I'm assuming this should be accepted syntax because the MDX Playground shows that the `{' '}` will result in a space:

![Screenshot 2024-08-07 at 18 28 47](https://github.com/user-attachments/assets/4c7cadbe-6632-46f0-a1f8-7dd9ac7e296b)

### Actual behavior

Problem is reported, as above

### OS

macOS

**Comment by ChristianMurphy** [comment]
@karlhorky the expression does indeed look like it isn't needed, it could be a space.
Is you concern with the word "unused"? Do you see this more as being a different rule, like "no-useless-expression"?

**Comment by karlhorky** [comment]
My thought here is that this isn't actually invalid syntax - it's a valid style in case a user would want to use it (eg. for our projects, we would like to use a space in JSX expression interpolation to avoid formatting problems with Prettier, rather than taking the route of disabling Prettier)

Stylistically, I can see how this is arguably an interesting rule for some users, but I would suggest that it doesn't absolutely need to be default
